parser-typechecker/src/Unison/Hashing/V2/Branch.hs

```diff
@@ -21,6 +20,11 @@ newtype MdValues = MdValues (Set MetadataValue)
   deriving (Eq, Ord, Show)
   deriving (Hashable) via Set MetadataValue
 
+newtype NameSegment = NameSegment Text deriving (Eq, Ord, Show)
+
+hashBranch :: Raw -> Hash
```

Comment:
Maybe `hashRawBranch` to avoid confusion with the other `hashBranch` and `hashBranch0`?

Reply:
Hm I think it's okay, this is part of a pretty different interface, and I wouldn't anticipate ever importing both packages.

But also, the fact that the type defined in this file is called `Raw` is something of a historical artifact, and could just as easily be named `Branch` as well. I think the function here is named more after the module and the concept than the data type.